flytekit/annotated/base_task.py

```diff
@@ -234,7 +236,11 @@ def __call__(self, *args, **kwargs):
             with ctx.new_execution_context(
                 mode=ExecutionState.Mode.LOCAL_TASK_EXECUTION, execution_params=new_user_params
             ):
-                return self.execute(**kwargs)
+                execute = self.execute
```

Comment:
couple comments as we discussed

1. does this work for all inputs / outputs (complex types)
2. Should we do this after the translation to Flyte types?
3. Does this workflow for both workflow and single task execution locally?

Eventually I would prefer if we use a common directory like
`~/.flyte/cache`
and for executions
`~/.flyte/executions`

@wild-endeavor ^